snuba/web/views.py

```diff
@@ -418,6 +418,47 @@ def snql_dataset_query_view(*, dataset: Dataset, timer: Timer) -> Union[Response
         assert False, "unexpected fallthrough"
 
 
+def _sanitize_payload(payload: Dict[str, Any], res: Dict[str, Any]) -> None:
+    def hex_encode_if_bytes(value: Any) -> Any:
+        if isinstance(value, bytes):
+            try:
+                return value.decode("utf-8")
+            except UnicodeDecodeError:
+                # encode the byte string in a hex string
+                return "RAW_BYTE_STRING__" + value.hex()
+
+        return value
+
+    for k, v in payload.items():
+        if isinstance(v, dict):
+            res[hex_encode_if_bytes(k)] = {}
+            _sanitize_payload(v, res[hex_encode_if_bytes(k)])
+        elif isinstance(v, list):
+            res[hex_encode_if_bytes(k)] = []
+            for item in v:
+                if isinstance(item, dict):
+                    res[hex_encode_if_bytes(k)].append({})
+                    _sanitize_payload(item, res[hex_encode_if_bytes(k)][-1])
+                else:
+                    res[hex_encode_if_bytes(k)].append(hex_encode_if_bytes(item))
+        else:
+            res[hex_encode_if_bytes(k)] = hex_encode_if_bytes(v)
+
+
+def dump_payload(payload: Dict[str, Any]) -> str:
+    try:
+        return json.dumps(payload, default=str)
```

Comment:
Doesn't adding `ensure_ascii=True` to this do the same thing as the `_sanitize_payload` function?

Reply:
no it doesn't. it still fails with 

```
FAILED tests/web/test_views.py::test_response_dumping - UnicodeDecodeError: 'utf-8' codec can't decode byte 0x83 in position 2: invalid start byte
```